pkg/service/article_service.go

```diff
@@ -0,0 +1,109 @@
+package service
+
+import (
+	"fmt"
+	"news-aggregator/pkg/model"
+	"news-aggregator/pkg/storage"
+	"time"
+)
+
+const (
+	ABCNewsSource         = "ABC News: International"
+	BBCNewsSource         = "BBC News"
+	WashingtonTimesSource = "The Washington Times stories: World"
+	NBCNewsSource         = "NBC News"
+	USATodaySource        = "USA TODAY"
+)
+
+//go:generate mockgen -destination=../service/mocks/mock_article_service.go -package=mocks news-aggregator/pkg/service Article
+
+type ArticleService struct {
+	articleStorage storage.ArticleStorage
+}
+
+// Article is an interface that defines the methods for interacting with the article storage.
+type Article interface {
```

Comment:
honestly i don't understand what's the main idea of this layer. It just calls storage. If so why do we need it?